### PR TITLE
Add test 'CRUD order return status'

### PR DESCRIPTION
**Pull request by nesrineabdmouleh**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Add test 'CRUD order return status'
| Type?         | refacto
| Category?     | TE
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | no
| How to test?  | TEST_PATH="functional/BO/13_shopParameters/02_orderSettings/02_statuses/03_CRUDOrderReturnStatus.js" URL_FO=yourShopURL npm run specific-test

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/21528)

### CODE REVIEW

File: tests/UI/pages/BO/shopParameters/orderSettings/statuses/index.js [hunk 1]
```diff
@@ -9,62 +9,71 @@ class Statuses extends BOBasePage {
...
 
     // Table body selectors
-    this.tableBody = `${this.gridTable} tbody`;
-    this.tableBodyRows = `${this.tableBody} tr`;
-    this.tableBodyRow = row => `${this.tableBodyRows}:nth-child(${row})`;
-    this.tableBodyColumn = row => `${this.tableBodyRow(row)} td`;
+    this.tableBody = tableName => `${this.gridTable(tableName)} tbody`;
+    this.tableBodyRows = tableName => `${this.tableBody(tableName)} tr`;
+    this.tableBodyRow = (tableName, row) => `${this.tableBodyRows(tableName)}:nth-child(${row})`;
+    this.tableBodyColumn = (tableName, row) => `${this.tableBodyRow(tableName, row)} td`;
```
Comment (Progi1984) [inline tests/UI/pages/BO/shopParameters/orderSettings/statuses/index.js]:
```suggestion
    this.tableBodyColumns = (tableName, row) => `${this.tableBodyRow(tableName, row)} td`;
```

File: tests/UI/pages/BO/shopParameters/orderSettings/statuses/index.js [hunk 2]
```diff
@@ -9,62 +9,71 @@ class Statuses extends BOBasePage {
...
 
     // Columns selectors
-    this.tableColumn = (row, idColumn) => `${this.tableBodyColumn(row)}:nth-child(${idColumn})`;
+    this.tableColumn = (tableName, row, idColumn) => `${this.tableBodyColumn(tableName, row)}:nth-child(${idColumn})`;
```
Comment (Progi1984) [inline tests/UI/pages/BO/shopParameters/orderSettings/statuses/index.js]:
```suggestion
    this.tableBodyColumn = (tableName, row, column) => `${this.tableBodyColumn(tableName, row)}:nth-child(${column})`;
```

**Review by Progi1984 — COMMENTED** [review]
Some comments

File: tests/UI/pages/BO/shopParameters/orderSettings/statuses/index.js (hunk 2 above)
Comment (nesrineabdmouleh) [reply tests/UI/pages/BO/shopParameters/orderSettings/statuses/index.js]: Done thx :+1:

File: tests/UI/pages/BO/shopParameters/orderSettings/statuses/index.js [hunk 3]
```diff
@@ -144,39 +166,42 @@ class Statuses extends BOBasePage {
    * @param page
    * @param row
    * @param columnName
-   * @param idColumn
+   * @param column
+   * @param tableName
    * @return {Promise<string>}
    */
-  async getTextColumn(page, row, columnName, idColumn) {
+  async getTextColumn(page, row, columnName, column, tableName = 'order') {
```
Comment (boubkerbribri) [inline tests/UI/pages/BO/shopParameters/orderSettings/statuses/index.js]: I think tableName should be the second parameter like in other pages and without a default value (tests that already exist should be fixed too)

```suggestion
  async getTextColumn(page, tableName, row, columnName, column) {
```

File: tests/UI/pages/BO/shopParameters/orderSettings/statuses/index.js [hunk 4]
```diff
@@ -144,39 +166,42 @@ class Statuses extends BOBasePage {
...
-   * Go to edit order status page
+   * Go to edit page
    * @param page
    * @param row
+   * @param tableName
    * @return {Promise<void>}
    */
-  async gotoEditOrderStatusPage(page, row) {
-    await this.clickAndWaitForNavigation(page, this.tableColumnActionsEditLink(row));
+  async gotoEditPage(page, row, tableName = 'order') {
```
Comment (boubkerbribri) [inline tests/UI/pages/BO/shopParameters/orderSettings/statuses/index.js]: same here

File: tests/UI/pages/BO/shopParameters/orderSettings/statuses/index.js [hunk 5]
```diff
@@ -144,39 +166,42 @@ class Statuses extends BOBasePage {
...
 
   /**
    * Delete order status from row
    * @param page
    * @param row
+   * @param tableName
    * @return {Promise<string>}
    */
-  async deleteOrderStatus(page, row) {
+  async deleteOrderStatus(page, row, tableName = 'order') {
```
Comment (boubkerbribri) [inline tests/UI/pages/BO/shopParameters/orderSettings/statuses/index.js]: same here

File: tests/UI/pages/BO/shopParameters/orderSettings/statuses/index.js [hunk 6]
```diff
@@ -189,45 +214,49 @@ class Statuses extends BOBasePage {
...
    * Select pagination limit
    * @param page
    * @param number
+   * @param tableName
    * @returns {Promise<string>}
    */
-  async selectPaginationLimit(page, number) {
-    await this.waitForSelectorAndClick(page, this.paginationDropdownButton);
-    await this.clickAndWaitForNavigation(page, this.paginationItems(number));
+  async selectPaginationLimit(page, number, tableName = 'order') {
```
Comment (boubkerbribri) [inline tests/UI/pages/BO/shopParameters/orderSettings/statuses/index.js]: same here

**Review by SimonGrn — DISMISSED** [review]

**Review by Progi1984 — DISMISSED** [review]

**Review by SimonGrn — APPROVED** [review]

**Review by Progi1984 — APPROVED** [review]

**Review by Progi1984 — APPROVED** [review]

**Pull request merged**